<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>Promise&async/await</title>
</head>
<body>
  コンソールに結果が表示されています。
  <script>
    // ===== Promiseで記述した場合 =====
    const getGitUsername = () => {
      return new Promise((resolve, reject) => {
        
        const url = 'https://api.github.com/users/wd30gsrc';
      
        // Github API をFetchメソッドで実行
        fetch(url).then(response => response.json())
          .then(json => {
          console.log('これは非同期処理成功時のメッセージです。');
          return resolve(json.login)
        }).catch(error => {
          console.log('これは非同期処理失敗のメッセージです。', error);
          return reject(null)
        })
        
      })
    };

    const message = 'GitのユーザーIDは'
    // const username = getGitUsername()
    // console.log(message + username)
    getGitUsername().then(username => {
      console.log(message + username)
    })

    // ===== async&awaitで記述した場合 =====
    const getGitUsername2 = async () => {
    const message = 'GitのユーザーIDは';
    const url = 'https://api.github.com/users/wd30gsrc';
    
    const json = await fetch(url)
      .then(response => {
        console.log('これは非同期処理成功時のメッセージ2です。')
        return response.json()
      }).catch(error => {
        console.log('これは非同期処理失敗時のメッセージ2です。', error)
        return null
      });
    console.log(message + json.login)
  }

  getGitUsername2();
  </script>
</body>
</html>